compute_endpoint/globus_compute_endpoint/endpoint/config/utils.py

```diff
@@ -200,6 +200,9 @@ def load_user_config_template(endpoint_dir: pathlib.Path) -> tuple[str, dict | N
     from globus_compute_endpoint.endpoint.endpoint import Endpoint
 
     user_config_path = Endpoint.user_config_template_path(endpoint_dir)
+    if not user_config_path.exists():
+        # Try without the .j2 extension
+        user_config_path = endpoint_dir / str(user_config_path).replace(".j2", "")
```

Comment:
Suggest doing a slice at the end (`str(user_config_path)[:-3]`) instead, since `some-ep.j2-name` is a valid (if insane) endpoint name

Reply:
Heh, I came here to observe that we implement `Endpoint.user_config_template_path()` as:

```python
@staticmethod
def user_config_template_path(endpoint_dir: pathlib.Path) -> pathlib.Path:
    return endpoint_dir / "user_config_template.yaml.j2"
```

and thus `.j2` is entirely in our control.  But in spelling it out for ~you~ myself, I now see that your point is not about our implementation but about the `endpoint_dir` which _could_ have a `.j2` in.  Ugh.  Heh.  "You win."  :laughing: 

("if inane".  "if insane".  Heh)

Reply:
Actually, does this work as expected?  As I read this:

```python
user_config_path = Endpoint.user_config_template_path(endpoint_dir)
   # == /home/me/.globus_compute/my_ep/user_config_template.yaml.j2
...
user_config_path = endpoint_dir / str(user_config_path).replace(".j2", "")
   # == /home/me/.globus_compute/my_ep/home/me/.globus_compute/my_ep/user_config_template.yaml
```

Meanwhile, I observe that this method previously made the assumption that the file exists.  Given that, suggest a log line.  Something like:

```python
user_config_path = Endpoint.user_config_template_path(endpoint_dir)
if not user_config_path.exists():
    log.info("Please update your path ... or something appropriate")
    user_config_path = pathlib.path(str(user_config_path)[:-3]))
```

Frankly, it's likely that _we'll_ be the ones to read that line.